hugegraph-core/src/main/java/com/baidu/hugegraph/backend/store/raft/RaftBackendStore.java

```diff
@@ -215,12 +216,19 @@ private Object queryByRaft(Query query, Function<Object, Object> func) {
         this.node().waitLeaderElected();
         StoreCommand command = new StoreCommand(StoreAction.QUERY);
         StoreClosure closure = new StoreClosure(command);
-        RaftNode raftNode = this.node();
-        raftNode.node().readIndex(BytesUtil.EMPTY_BYTES, new ReadIndexClosure() {
+        ReadIndexClosure readIndexClosure = new ReadIndexClosure() {
             @Override
             public void run(Status status, long index, byte[] reqCtx) {
                 if (status.isOk()) {
                     closure.complete(status, () -> func.apply(query));
+                } else if (status.getRaftError() == RaftError.EAGAIN) {
+                    LOG.warn("Failed to execute query {} because of leader " +
+                             "has not committed any log entry at its term",
+                             query);
+                    closure.failure(status, new RaftException(
+                            "Failed to execute query '%s' because of leader " +
+                            "has not committed any log entry at its term",
```

Comment:
move to "status.getRaftError() == RaftError.EAGAIN" block to line 247, just throw exception in else block